core/src/main/java/com/linecorp/armeria/common/Flags.java

```diff
@@ -1087,27 +1089,37 @@ public static boolean validateHeaders() {
 
     @Nullable
     private static String nullableCaffeineSpec(String name, String defaultValue) {
-        return getCaffeineSpec(name, defaultValue, true);
+        return caffeineSpec(name, defaultValue, true);
     }
 
-    private static String caffeineSpec(String name, String defaultValue) {
-        return requireNonNull(getCaffeineSpec(name, defaultValue, false), "caffeineSpec");
+    private static String nonnullCaffeineSpec(String name, String defaultValue) {
+        return requireNonNull(caffeineSpec(name, defaultValue, false));
```

Comment:
Don't we need to specify `message` for `requireNonNull` such as `requireNonNull(..., name)`?

Reply:
It never returns null when `allowOff` is `false`. `requireNonNull` here is just to make the static analysis happy.

Reply:
Changed to use `assert`